### PR TITLE
Making state machines global in React

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+author:
+  - David Khourshid
+originalURL: ""
+excerpt: ""
+publishedAt: 2023-1-27
+---
+
+React developers often need to share state between components. While the `useMachine(...)` hook provides a convenient way to represent local state as a state machine, it’s not very feasible for shared or global state. Thankfully, the `createActorContext(machine)` function from `@xstate/react` (released in `@xstate/react@3.1.0`) is a convenient way to share state machines globally in any React application.
+
+The `createActorContext(machine)` function returns a [React Context object](https://beta.reactjs.org/learn/passing-data-deeply-with-context) that interprets the machine and makes the actor (interpreted machine) available through React Context. It has helper methods for accessing state and the actor ref. It takes one argument, `machine`, which is a state machine from `createMachine(...)` or a function that lazily returns a machine.
```
Comment (laurakalbag) [inline content/posts/2023-1-27-making-state-machines-global-in-react.mdx:17]: Do the `it`s here in `It has helper methods` and `It takes one argument` refer to `createActorContext`, the actor, or the React Context object?

**Review by laurakalbag — COMMENTED** [review]
This is great, concise and easy to understand. Just one question in the suggestions where I wasn’t sure which “it”s you were referring to…

Also, do you think this post is long enough to justify having a couple of headings to help make the most more skimmable?

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx (hunk 1 above)
Comment (davidkpiano) [reply content/posts/2023-1-27-making-state-machines-global-in-react.mdx:18]: It refers to `createActorContext(...)`

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      <button onClick={() => actorRef.send({ type: "INCREMENT" })}>
+        Increment
+      </button>
+    </div>
+  );
+}
+```
+
+Lastly, if you need to provide implementation details to the machine, such as `actions`, `guards`, or `delays`, you can provide a similar machine in the `machine={...}` prop on the `Provider` component:
+
```
Comment (kevinmaes) [inline content/posts/2023-1-27-making-state-machines-global-in-react.mdx:82]: What do we mean by "similar" machine? What happens if I try to pass a completely different machine?

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx (hunk 1 above)
Comment (davidkpiano) [reply content/posts/2023-1-27-making-state-machines-global-in-react.mdx]: Clarified 👍

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      <SomeComponent />
+    </SomeMachineContext.Provider>
+  );
+}
+```
+
+To consume the actor in a component, you can use the `useActor()` hook to get the `state` and `send` function, or you can use `useSelector(selector, compare?)` hook to derive a specific value from the snapshot:
+
+```tsx
+import { SomeMachineContext } from "../path/to/SomeMachineContext";
```
[the remark this answers is not in the document]
Comment (davidkpiano) [reply content/posts/2023-1-27-making-state-machines-global-in-react.mdx]: Added filenames 👍

**Comment by davidkpiano** [comment]
@laurakalbag Added the Youtube link and headings. Feel free to revise them ✍️

File: content/posts/2023-1-27-making-state-machines-global-in-react.mdx (hunk 2 above)
Comment (kevinmaes) [reply content/posts/2023-1-27-making-state-machines-global-in-react.mdx:94]: You answered this a bit in the livestream. Sounds like it needs to be the same machine and that typegen will warn against what it identifies as not being the same machine. Is that correct?

**Comment by laurakalbag** [comment]
> @laurakalbag Added the Youtube link and headings. Feel free to revise them ✍️

No need, this is perfect!

**Pull request merged**